extractor/extractor.go

```diff
@@ -0,0 +1,266 @@
+package extractor
+
+import (
+	"bufio"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/vision-cli/common/cases"
+	"github.com/vision-cli/common/transpiler/model"
+	protoparser "github.com/yoheimuta/go-protoparser"
+	parserStructs "github.com/yoheimuta/go-protoparser/parser"
+)
+
+func ProjectStructure(projectDirectory string) ([]model.Module, error) {
+	targetDir := filepath.Join(projectDirectory, "services")
+
+	// Get module names and api versions
+	modules, err := Modules(targetDir)
+	if err != nil {
+		return nil, fmt.Errorf("error reading modules: %w", err)
+	}
+
+	// Get services for each module
+	for i, module := range modules {
+		moduleNameWithVersion := fmt.Sprintf("%s.%s", module.Name, module.ApiVersion)
+		moduleDirectory := filepath.Join(targetDir, moduleNameWithVersion)
+		modules[i].Services, err = Services(moduleDirectory, &module)
+		if err != nil {
+			return nil, fmt.Errorf("error reading services: %w", err)
+		}
+	}
+
+	return modules, nil
+}
+
+func Modules(targetDir string) ([]model.Module, error) {
+	moduleDirs, err := os.ReadDir(targetDir)
+	if err != nil {
+		return nil, fmt.Errorf("error reading directory, please ensure project root is correct: %w", err)
+	}
+
+	modules := make([]model.Module, 0, len(moduleDirs))
+
+	for _, path := range moduleDirs {
+		if path.IsDir() && strings.Contains(path.Name(), ".") {
+			parts := strings.Split(path.Name(), ".")
+			if len(parts) == 2 {
+				modules = append(modules, model.Module{
+					Name:       parts[0],
+					ApiVersion: parts[1],
+				})
+			}
+		}
+	}
+
+	return modules, nil
+}
+
+func Services(moduleDirectory string, module *model.Module) ([]model.Service, error) {
+	serviceDirs, err := os.ReadDir(moduleDirectory)
+	if err != nil {
+		return nil, fmt.Errorf("error reading module directory: %w", err)
+	}
+
+	services := []model.Service{}
+
+	for i, path := range serviceDirs {
+		if path.IsDir() {
+			serviceName := path.Name()
+			services = append(services, model.Service{Name: serviceName})
+
+			// Construct the .proto file path
+			protoFilename := fmt.Sprintf("%s_%s%s%s", module.Name, module.ApiVersion[:1], module.ApiVersion[1:], serviceName)
+			protoFilename = cases.Snake(protoFilename)
+			protoFilename = fmt.Sprintf("%s.proto", protoFilename)
+			protoFilePath := filepath.Join(moduleDirectory, serviceName, "proto", protoFilename)
+
+			services[i].Enums, err = getEnums(protoFilePath)
+			if err != nil {
+				return nil, fmt.Errorf("error getting enums: %w", err)
+			}
+
+			modelsFolder := filepath.Join(moduleDirectory, path.Name(), "models", "models.go")
```

Comment:
This seems more like the path to the models file rather than the folder.

Reply:
agree, pls rename to something like "modelsFile"